### PR TITLE
[Accuracy diff No.117] Fix accuracy diff for paddle.nn.functional.dice_loss API

### PR DESCRIPTION
**Pull request by NKNaN**
根据 paddle 的实现修改 DiceLossRule

回测结果：
image

### CODE REVIEW

**Review by cangtianhuang — APPROVED** [review]
great using one-hot, LGTM!
